<?xml version="1.0"?>
<launch>
  <!-- rviz -->
    <node pkg="rviz" type="rviz" name="fusion_rviz" required="true" 
      args = "-d $(find fusion_nodelet)/rviz/fusion.rviz" />
  <!-- urdf -->
    <arg name="gui" default="False" />
    <param name="robot_description" textfile="$(find fusion_nodelet)/urdf/fusion.urdf" />
    <param name="use_gui" value="$(arg gui)"/>
    <!-- <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" /> -->
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher" />
  <!-- rosbag play -->
    <node pkg="rosbag" type="play" name="fusion_bag_play" args="-l -r 1.0 $(env HOME)/data/bag/2018-12-17-15-58-31.bag" />
  
  <!-- nodelet*************nodelet -->

  <!-- nodelet::Global Arg -->
    <env name="ROSCONSOLE_CONFIG_FILE" value="$(find fusion_nodelet)/config/lidar_perception.config"/>
    <arg name="arg_nodelet_manager" default="velodyne_nodelet_manager" />
    <arg name="arg_pose_path" default="$(find fusion_nodelet)/config/calib_lidar.txt"/>

  <!-- nodelet -->
    <node pkg="nodelet" type="nodelet" name="$(arg arg_nodelet_manager)" args="manager" output="screen"/>
  <!-- nodelet::FusionCloudNodelet -->
    <node pkg="nodelet" type="nodelet" name="FusionCloudNodelet" args="load fusion_nodelet/FusionCloudNodelet $(arg arg_nodelet_manager)">
      <param name="trans_file_path" value="$(find fusion_nodelet)/config/calib_lidar.txt" />
    </node>
    
  <!-- nodelet::PerceptionNodelet -->
    <node pkg="nodelet" type="nodelet" name="PerceptionNodelet" args="load fusion_nodelet/PerceptionNodelet $(arg arg_nodelet_manager)">
      <param name="perception_cloud_topic" value="/apollo/sensor/velodyne16/compensator/PointCloud2"/>
      <!-- <param name="h_lidar" value="1"/> -->
    </node>
</launch>
